python/mxnet/image/detection.py

```diff
@@ -0,0 +1,611 @@
+# pylint: disable=unused-import
+"""Read images and perform augmentations for object detection."""
+
+from __future__ import absolute_import, print_function
+
+import random
+import logging
+import numpy as np
+
+from ..base import numeric_types
+from .. import ndarray as nd
+from .._ndarray_internal import _cvcopyMakeBorder as copyMakeBorder
+from .. import io
+from .image import RandomOrderAug, ColorJitterAug, LightingAug, ColorNormalizeAug
+from .image import ResizeAug, ForceResizeAug, CastAug, fixed_crop, ImageIter
+
+
+def BorrowAug(augmenter):
+    """Borrow standard augmenter from image classification
+    which is good if label won't be affected."""
+
+    def aug(src, label):
+        """Augmenter body"""
+        src = augmenter(src)[0]
+        return (src, label)
+
+    return aug
+
+
+def DetHorizontalFlipAug(p):
+    """Random horizontal flippling."""
+
+    def horizontal_flip_label(label):
+        """Helper function to flip label."""
+        tmp = 1.0 - label[:, 3]
+        label[:, 1] = tmp
+        label[:, 3] = tmp
+
+    def aug(src, label):
+        """Augmenter body"""
+        if random.random() < p:
+            src = nd.flip(src, axis=1)
+            # flip ground-truths as well
+            horizontal_flip_label(label)
+        return (src, label)
+
+    return aug
+
+def DetRandomCropAug(p, min_object_covered=0.1, aspect_ratio_range=(0.75, 1.33),
```

Comment:
This looks pretty complicated. Please add some tests and document.

explain what this does in some detail and explain the arguments one by one.
See other functions for example